src/lib.rs

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use astarte_sdk::AstarteSdk;
+pub mod error;
+pub mod hardwareinfo;
+pub mod osinfo;
+
+use astarte_sdk::builder::AstarteOptions;
+use clap::Parser;
+use error::DeviceManagerError;
+
+#[derive(Debug, Parser)]
+struct Cli {
+    // Realm name
+    #[clap(short, long)]
+    realm: String,
+    // Device id
+    #[clap(short, long)]
+    device_id: String,
+    // Credentials secret
+    #[clap(short, long)]
+    credentials_secret: String,
+    // Pairing URL
+    #[clap(short, long)]
+    pairing_url: String,
+}
+
+pub struct DeviceManager {
+    sdk: AstarteSdk,
+}
+
+impl DeviceManager {
+    pub async fn new() -> Result<DeviceManager, DeviceManagerError> {
+        let Cli {
+            realm,
+            device_id,
+            credentials_secret,
+            pairing_url,
+        } = Parser::parse();
+
+        let sdk_options =
+            AstarteOptions::new(&realm, &device_id, &credentials_secret, &pairing_url)
+                .interface_directory("./interfaces")?
```

Comment:
same here, let's define somewhere a `interface_json_path` instead of hardcoding.